bpf/lib/nat.h

```diff
@@ -1514,6 +1521,27 @@ snat_v6_needs_masquerade(struct __ctx_buff *ctx __maybe_unused,
 		}
 	}
 
+/* Check if the packet matches an egress NAT policy and so needs to be SNAT'ed.
+ *
+ * This check must happen before the IPV4_SNAT_EXCLUSION_DST_CIDR check below as
+ * the destination may be in the SNAT exclusion CIDR but regardless of that we
+ * always want to SNAT a packet if it's matched by an egress NAT policy.
+ */
+#if defined(ENABLE_EGRESS_GATEWAY_COMMON)
+	if (egress_gw_snat_needed_hook_v6(tuple->saddr, tuple->daddr, &target->addr,
+				       &target->ifindex)) {
+		if (target->addr == EGRESS_GATEWAY_NO_EGRESS_IP_V6)
+			return DROP_NO_EGRESS_IP;
+
+		target->egress_gateway = true;
+		/* If the endpoint is local, then the connection is already tracked. */
+		if (!local_ep)
+			target->needs_ct = true;
+
+		return NAT_NEEDED;
+	}
+#endif
```

Comment:
This is in the egress path on the gateway node, as the packet (local or redirected in `from-overlay`) ends up in to-netdev on the egress interface.. We apply SNAT with the desired `egressIP`.